server/sonar-server/src/main/java/org/sonar/server/platform/Platform.java

```diff
@@ -84,38 +100,50 @@ protected void doStart(Startup startup) {
       return;
     }
 
-    boolean wentToSafemode = false;
-    if (requireSafeMode()) {
-      LOGGER.info("DB needs migration, entering safe mode");
-      wentToSafemode = true;
-      startSafeModeContainer();
-      currentLevel = levelSafeMode;
-      started = true;
-    }
-    // if AutoDbMigration kicked in, safe mode is not required anymore
-    if (!requireSafeMode()) {
-      if (wentToSafemode) {
-        LOGGER.info("DB has been updated, exiting safe mode");
-      }
-      startLevel34Containers();
-      executeStartupTasks(startup);
-      // switch current container last to avoid giving access to a partially initialized container
-      currentLevel = level4;
-      started = true;
-
-      // stop safemode container if it existed
-      stopSafeModeContainer();
+    boolean dbRequiredMigration = dbRequiresMigration();
+    startSafeModeContainer();
+    currentLevel = levelSafeMode;
+    started = true;
+
+    // if AutoDbMigration kicked in or no DB migration was required, startup can be resumed in another thread
+    if (dbRequiresMigration()) {
+      LOGGER.info("Database needs migration");
+    } else {
+      this.autoStarter = autoStarterSupplier.get();
+      this.autoStarter.execute(() ->  {
+        try {
+          if (dbRequiredMigration) {
+            LOGGER.info("Database has been automatically updated");
+          }
+          startLevel34Containers();
+
+          executeStartupTasks(startup);
+          // switch current container last to avoid giving access to a partially initialized container
+          currentLevel = level4;
+
+          // stop safemode container if it existed
+          stopSafeModeContainer();
+        } catch (Throwable t) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1181)